meshack-mbuvi/ride-my-way-api
# #158521474 Endpoint to create new ride offer

**What does this PR do?
Adds endpoint to create new ride offer

**Description of Task to be completed?**
- Create an `POST /api/v1/rides/` endpoint to add new ride offer
- Add model for a single ride offer
- Add view function to create a new offer using the development model for the ride offer
- Document this new endpoint on README.md

**How should this be manually tested?**
- Clone this repository
- In your terminal, cd ride-my-way-api
- `git checkout ft-create-ride-158521474
- Activate virtual environment and install the dependancies
- Execute `python3 run.py`
- Using 'Postman', point to 'your-server:5000/api/v1/rides with json data in the format:
```
data= {
    "destination": "Nairobi",
    "route": "Thika",
    "start_point": "Thika",
    "start_time": "June 01 1990 09:00AM",
    "available_space": 10}
}
```

The response should look like:
```
{
   'message': 'ride offer added successfully.',
   'offer id': {offer_id}
}
```

**What are the relevant pivotal tracker stories?**
[#158521474](https://www.pivotaltracker.com/story/show/158521474)

**Screenshots**
![create ride](https://user-images.githubusercontent.com/9263906/41822830-7515d9d6-77fe-11e8-824e-ef8b12e8b7fe.png)
![rideoffer](https://user-images.githubusercontent.com/9263906/41822861-e7672ada-77fe-11e8-8b61-d6d4b700b2ca.png)
